data/data-pipeline/data_pipeline/etl/sources/census_acs/etl_imputations.py

```diff
@@ -40,24 +42,44 @@ def _choose_best_mask(
 def _prepare_dataframe_for_imputation(
     impute_var_named_tup_list: List[NamedTuple],
     geo_df: gpd.GeoDataFrame,
+    population_field: str,
+    minimum_population_required_for_imputation: int = 1,
     geoid_field: str = "GEOID10_TRACT",
 ) -> Tuple[Any, gpd.GeoDataFrame]:
+    """Helper for imputation.
+
+    Given the inputs of `ImputeVariables`, returns list of tracts that need to be
+    imputed, along with a GeoDataFrame that has a column with the imputed field
+    "primed", meaning it is a copy of the raw field.
+
+    Will drop any rows with population less than
+    `minimum_population_required_for_imputation`.
+    """
     imputing_cols = [
         impute_var_pair.raw_field_name
         for impute_var_pair in impute_var_named_tup_list
     ]
 
-    # prime column to exist
+    # Prime column to exist
     for impute_var_pair in impute_var_named_tup_list:
         geo_df[impute_var_pair.imputed_field_name] = geo_df[
             impute_var_pair.raw_field_name
         ].copy()
 
-    # generate a list of tracts for which at least one of the imputation
-    # columns is null
-    tract_list = geo_df[geo_df[imputing_cols].isna().any(axis=1)][
-        geoid_field
-    ].unique()
+    # Generate a list of tracts for which at least one of the imputation
+    # columns is null that also meets population criteria.
+    tract_list = geo_df[
+        (
+            # First, check whether any of the columns we want to impute contain null
+            # values
+            geo_df[imputing_cols].isna().any(axis=1)
+            # Second, ensure population is either null or >= the minimum population
+            & (
+                geo_df[population_field].isnull() | geo_df[population_field]
```

Comment:
Because of the way black splits the line, I'd like the inequality comparison wrapped in another set of parent to help signal "wait there's more to this)

Reply:
nice!